### PR TITLE
Interaction

### PR DESCRIPTION
**Pull request by fdansv**
(Finally got it right @alonsogarciapablo ;) )

### CODE REVIEW

File: examples/magicexample.html [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    tiler_template: "http://development.localhost.lan:8181"
+  }).addTo(map);
+
+  var alternate = "/** choropleth visualization */ #ne_10m_populated_places_simple_6{ marker-fill-opacity: 0.8; marker-line-color: #FFF; marker-line-width: 1; marker-line-opacity: 1; marker-width: 10; marker-fill: #FFFFCC; marker-allow-overlap: true; } #ne_10m_populated_places_simple_6 [ pop_max <= 35676000] { marker-fill: #0C2C84; } #ne_10m_populated_places_simple_6 [ pop_max <= 534555] { marker-fill: #225EA8; } #ne_10m_populated_places_simple_6 [ pop_max <= 211138] { marker-fill: #1D91C0; } #ne_10m_populated_places_simple_6 [ pop_max <= 109023] { marker-fill: #41B6C4; } #ne_10m_populated_places_simple_6 [ pop_max <= 59575] { marker-fill: #7FCDBB; } #ne_10m_populated_places_simple_6 [ pop_max <= 31253] { marker-fill: #C7E9B4; } #ne_10m_populated_places_simple_6 [ pop_max <= 11739] { marker-fill: #FFFFCC; }";
+
+  cartoDBd3Layer.on(0, "featureOut", function(f, d3selection){
+    d3selection.reset();
+  });
+
+  cartoDBd3Layer.on(0, "featureOver", function(f, d3selection){
```
Comment (pabletecodes) [inline examples/magicexample.html:123]: In CartoDB.js v3, we had something more like this:

```
cartoDBd3Layer.on("featureOver", function(event, latlng, pos, data, layer) {
  // ...
});
```

I'm not saying we should follow that route cause we can also change this in v4, but we should be consistent in both cases.

Comment (fdansv) [inline examples/magicexample.html:123]: @alonsogarciapablo yeah but note that cartoDBd3Layer can contain more than one layer, hence the index

File: src/leaflet_d3.js [hunk 2]
```diff
@@ -19,6 +19,15 @@ L.CartoDBd3Layer = L.Class.extend({
     L.Util.setOptions(this, options)
   },
 
+  on: function(index, eventName, callback) {
+    if (eventName in this.renderers[index].events){
+      this.renderers[index].on(eventName, callback);
+    }
```
Comment (pabletecodes) [inline src/leaflet_d3.js]: According to standardjs.com, we should write ifs like:

```
if (somethingIsTrue) {
  // ...
} else {
  // ...
}
```

File: examples/magicexample.html (hunk 1 above)
Comment (pabletecodes) [inline examples/magicexample.html:123]: In the previous example I provided, the `layer` param of the callback is actually the index of the layer that was clicked/hovered. Basically, you bind events to the LayerGroup and then decide what do in the event callback.

**Comment by fdansv** [comment]
Just like we did with filtering, I'm merging this with just the public API pending.

**Pull request merged**